File: src/main/java/org/gecko/tools/PanTool.java
package org.gecko.tools;

import org.gecko.actions.ActionManager;

public class PanTool extends Tool {

    private static final String NAME = "Pan Tool";

    public PanTool(ActionManager actionManager) {
        super(actionManager);
    }

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String getIconPath() {
        //TODO stub
        return null;
    }
}

File: src/main/kotlin/org/gecko/tools/PanTool.kt
package org.gecko.tools

import javafx.scene.Cursor
import org.gecko.actions.ActionManager
import org.gecko.view.views.ViewElementPane

/**
 * A concrete representation of a pan-[Tool], utilized for moving the view.
 */
class PanTool(actionManager: ActionManager) : Tool(actionManager, ToolType.PAN, true) {
    override fun visitView(pane: ViewElementPane) {
        super.visitView(pane)
        pane.draw().isPannable = true
        pane.draw().cursor = Cursor.OPEN_HAND
    }
}
